src/categorical_algebra/CSets.jl

```diff
@@ -588,12 +596,22 @@ function backtracking_search(f, X::StructACSet{S}, Y::StructACSet{S};
   backtracking_search(f, state, 1)
 end
 
-function backtracking_search(f, state::BacktrackingState, depth::Int)
+function backtracking_search(f, state::BacktrackingState{S}, depth::Int) where {S}
   # Choose the next unassigned element.
   mrv, mrv_elem = find_mrv_elem(state, depth)
   if isnothing(mrv_elem)
     # No unassigned elements remain, so we have a complete assignment.
-    return f(ACSetTransformation(state.assignment, state.dom, state.codom))
+    if !isempty(state.loose) && any([f!=identity for f in state.loose])
+      ac = NamedTuple([
```

Comment:
Using [this constructor](https://github.com/AlgebraicJulia/Catlab.jl/blob/12a6dbed89186757bb0168f3d1f8d662cd7b5c2c/src/categorical_algebra/CSets.jl#L324), can replace everything inside the conditional with

```julia
f(LooseACSetTransformation{S}(state.assignment, state.loose, state.dom, state.codom)
```